### ISSUE TITLE
get much higher non_na_macro_f1 than the paper

### ISSUE THREAD

**Issue by koala-222** [Spico197/NYT-H#3]
hello, I have read your paper and thanks for your great work😁
But I have tried to run the bash scripts you provided, and I have a question that makes me confused:
When I run pcnn at sentence level, the hyper parameters are default as in your .sh file:

~~~bash
python -u run.py \
    --description="NYT-H training with pcnn on sentence-level" \
    --task_name="sent" \
    --model_name="sent_pcnn" \
    --data_dir="data/train" \
    --eval_model="best" \
    --overwrite_output_dir \
    --output_dir="outputs/sent_pcnn_422" \
    --embedding_path="/data/klhao/Glove/glove.6B.50d.txt" \
    --embedding_dim=50 \
    --seed=422 \
    --main_device="cuda:0" \
    --device_ids="[0]" \
    --epoch=50 \
    --batch_size=64 \
    --dropout_rate=0.5 \
    --lr=1e-3 \
    --max_len=150 \
    --pos_dis=75 \
    --pos_dim=5 \
    --save_best_model \
    --do_eval \
    --do_train \
    --select_score='non_na_macro_f1' \
    --tb_logging_step=1000
~~~

I got much higher results than you reported in the paper, the final test results is below:

~~~bash
[Wed 09 Dec 2020 15:19:10]-INFO-run.py-210-18063: Epoch: 50, DEV non_na_macro_f1: 0.5439261196000152
Epoch: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [15:41<00:00, 18.82s/it]
[Wed 09 Dec 2020 15:19:10]-INFO-run.py-283-18063: global step = 75650, average loss = 0.27621036564573026
[Wed 09 Dec 2020 15:19:10]-INFO-evaluate.py-85-18063: ***** Running evaluation  *****
[Wed 09 Dec 2020 15:19:10]-INFO-evaluate.py-86-18063:   Num examples = 9955
Evaluating: 100%|█████████| 156/156 [00:02<00:00, 62.77it/s]
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-345-18063: ***** test Eval results  *****
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   acc = 0.7980914113510799
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   auc = 0.7637563378273143
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   dsgt_f1 = 0.7198344419510546
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   dsgt_p = 0.7200986489756549
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   dsgt_r = 0.7230293803920519
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   macro-f1 = 0.7198344419510546
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   macro-precision = 0.7200986489756549
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   macro-recall = 0.7230293803920519
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   magt_f1 = 0.48830020158337806
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   magt_p = 0.4063063910846374
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   magt_r = 0.7191573395022106
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   max_f1 = 0.8036410923276983
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   mean_prec = 0.18658498965719159
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   micro-f1 = 0.7980914113510799
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   micro-precision = 0.7980914113510799
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   micro-recall = 0.7980914113510799
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   non_na_auc = 0.8969445947216447
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   non_na_macro_f1 = 0.7198344419510546
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   non_na_macro_precision = 0.7200986489756549
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   non_na_macro_recall = 0.7230293803920519
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   report = {'NA': {'precision': 0.0, 'recall': 0.0, 'f1-score': 0.0, 'support': 0}, '/business/company/founders': {'precision': 0.9583333333333334, 'recall': 0.9261744966442953, 'f1-score': 0.9419795221843005, 'support': 149}, '/people/person/ethnicity': {'precision': 1.0, 'recall': 1.0, 'f1-score': 1.0, 'support': 22}, '/location/administrative_division/country': {'precision': 0.9782178217821782, 'recall': 0.9821073558648111, 'f1-score': 0.9801587301587302, 'support': 503}, '/business/company/place_founded
[Wed 09 Dec 2020 15:20:28]-INFO-evaluate.py-86-18939:   Num examples = 9955
~~~

the non_na_macro_f1 is 0.7198, so much higher than 0.5768 in the paper; and the magt_f1 is 0.4883, also higher than 0.4401 in the paper.

I have merged train_nonna.json and na_train.json as train.json as  you mentioned.

Do I do something wrong? oh, and, besides pcnn, other baseline models at sentence level have higher f1 score than the paper. I don't know where is the problem. 
Hope for your reply~ Thank you.😄

**Comment by Spico197** [comment]
Hi, thanks for your attention on our work. Congrats for your higher results!

Sorry to hear about the reproduction problem. I've checked my training logs and find it's a hyperparameter-setting issue. The actual hyperparameters are slightly different among these models on different tracks, and we didn't do too much tuning work. For the `sent_pcnn` model results reported in the paper, the `batch_size` should be 256. I'll publish all the configuration files ASAP for easier results reproduction.

Besides, the results you provided also show the gaps between DSGT and MAGT scores, which also indicates the DS evaluation bias. Hope our NYT-H can be a possible solution for this problem, and warmly welcome to try it in different tracks.
